python/lsst/ts/standardscripts/maintel/m1m3/check_hardpoint.py

```diff
@@ -0,0 +1,146 @@
+
+        self.mtcs = MTCS(
+            domain=self.domain,
+            intended_usage=None if add_remotes else MTCSUsages.DryTest,
+            log=self.log,
+        )
+
+        for comp in self.mtcs.components_attr:
+            if comp != "mtm1m3":
+                setattr(self.mtcs.check, comp, False)
+
+        self.timeout_check = 60
+        self.timeout_std = 60
+
+        self.hardpoints = range(1, 7)
+
+    @classmethod
+    def get_schema(cls):
+        url = "https://github.com/lsst-ts/"
+        path = (
+            "ts_externalscripts/blob/main/python/lsst/ts/standardscripts/"
+            "maintel/m1m3/check_hardpoint.py"
+        )
+        schema_yaml = f"""
+        $schema: http://json-schema.org/draft-07/schema#
+        $id: {url}{path}
+        title: CheckHardpoint v1
+        description: Configuration for Maintel check hardpoint breakaway test SAL Script.
+        type: object
+        properties:
+            hardpoints:
+                description: Hardpoints to run the breakway test.
+                oneOf:
+                  - type: array
+                    minItems: 1
+                    items:
+                      type: number
+                      minimum: 1
+                      maximum: 6
+                  - type: string
+                    enum: ["all"]
+                default: "all"
+        additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        self.hardpoints = (
+            range(1, 7) if config.hardpoints == "all" else config.hardpoints
+        )
+
+    def set_metadata(self, metadata):
+        metadata.duration = self.timeout_check * 2 + self.timeout_std
+
+    async def wait_hardpoint_test_ok(self, hp):
+        """Wait for the hardpoint test to be passed."""
+
+        self.log.info("Checking if the hardpoint breakaway test has passed.")
+
+        while True:
+            test_status = await self.m1m3.evt_hardpointTestStatus.next(
+                flush=True, timeout=self.timeout_check
+            )
+            self.hardpoint_test_state = test_status.testState[hp - 1]
+            if self.hardpoint_test_state == HardpointTest.FAILED:
+                raise RuntimeError(f"Hardpoint {hp} test FAILED")
+            elif self.hardpoint_test_state == HardpointTest.PASSED:
+                self.log.info(f"Hardpoint {hp} test PASSED")
+                break
+
+    async def run(self):
+        # Check that the MTCS is in the right state
+        await asyncio.gather(
+            self.mtcs.assert_all_enabled(), self.mtcs.assert_liveliness()
+        )
+
+        await self.mtcs.enter_m1m3_engineering_mode()
+
+        for idx, hp in enumerate(self.hardpoints):
+            await self.checkpoint(f"Testing hardpoint {hp}.")
+            await self.mtcs.run_m1m3_hard_point_test(hp=hp)
+            self.mtcs.rem.mtm1m3.evt_hardpointTestStatus.flush()
```

Comment:
The `self.mtcs.run_m1m3_hard_point_test` already waits for the test to finish, you can remove all this other parts here.